### ISSUE TITLE
Unable to install the CEA Tool for ArcGIS

### ISSUE THREAD

**Issue by ahmed3yad** [architecture-building-systems/CEAforArcGIS#571]
Dear all,

I'm a researcher and I want to use the CEA tool for ArcGIS for my research. I tried to install it and followed the instructions but then It showed me an error while I was testing the example shipped with the tool (The radiation module did not work). So, I tried to remove the tool and install it again but now the tool is not even working and shows a red cross beside its logo in ArcGIS.

So, can you give me a link for the latest .exe setup file for the tool or at least tell me an easy way that I can follow to install the tool for ArcGIS (latest update) because I have no much experience in using python and the installation guide published on the website is really hard to follow.

Email: ahmed.m.ayad@outlook.com

Best Regards,

**Comment by daren-thomas** [comment]
@ahmed3yad we are currently spending a lot of effort to make the installation much less painful.
Please check this page for the a preview of how to install the latest version: https://github.com/architecture-building-systems/CEAforArcGIS/blob/master/docs/installation-guide.rst

**Comment by ahmed3yad** [comment]
@daren-thomas Thanks a lot for your fast response and I know of course that you spend a lot of effort to make this wonderful tool easy to use. 

So, in order to install the CEA tool for ArcGIS I should follow these 4 steps only?:
1) create a conda environment and activate it.
2) conda install -c conda-forge geopandas
3) pip install cityenergyanalyst
4) cea install-toolbox

Because there are two other steps mentioned in the web page (Connecting to Arcpy & Installation from GitHub repository) and I do not know If I should do them too or the above four steps are enough?

Thanks a lot!

**Comment by daren-thomas** [comment]
@ahmed3yad to quote from the page:

> The command cea install-toolbox attempts to connect the CEA with the ArcGIS environment. You should not need to do anything else.

You should not need to do anything else than the four steps above! But if you run into problems, you can check the "Connecting to Arcpy" section. If you wish to install the absolute bleeding edge version, then replace step 3 (`pip install cityenergyanalyst`) with a git clone of the CEAforArcGIS repository and then a `python setup.py install`. I will update the documentation to reflect this clarification.

The steps mentioned on the gitbooks page are obsolete and will be removed once we have finished transitioning to the new documentation (expected some time end of April).

**Comment by daren-thomas** [comment]
see PR #574

**Comment by ahmed3yad** [comment]
@daren-thomas I followed the four steps and replaced the third step with: "a git clone of the CEAforArcGIS repository and then a `python setup.py install`". But at the end, the tool showed under "My Toolboxes" with a red X beside it and no modules were installed as you can see in the following picture:

![capture](https://cloud.githubusercontent.com/assets/26455475/24510074/68aa1148-1568-11e7-9269-2a418147b50c.PNG)

Do you know what went wrong?

**Comment by daren-thomas** [comment]
@ahmed3yad, I am on it! I'm sorry. I made some stupid mistakes. I'll notify you when this is fixed.

**Comment by ahmed3yad** [comment]
@daren-thomas Great! ... Thanks a lot and waiting for your notification.

**Comment by ahmed3yad** [comment]
@daren-thomas Any news about the fix ? Thanks!

**Comment by daren-thomas** [comment]
@ahmed3yad as per pull request #582, the fix is now in the master branch. You can update your repository and run `python setup.py install` again. It should now work.

**Comment by ahmed3yad** [comment]
@daren-thomas Thanks Daren, but unfortunately the problem still exists and the tool is still not working !

**Comment by daren-thomas** [comment]
@ahmed3yad did you re-do the `cea install-toolbox` command?

**Comment by ahmed3yad** [comment]
@daren-thomas Yes, I installed everything from the beginning as I reinstalled windows, ArcGIS and Anaconda to make sure that everything will work as intended and then when I ran the command: “python setup.py install” I had an error that told me to install “Microsoft Visual C++ Compiler Package for Python 2.7” and when I installed it I had another error that required me to install “VCredist SP1” which I did.

After I finished solving these two problems, the “python setup.py install” step went fine but then when I moved to the “cea install-toolbox” step I had another error that told me that “ephem” is not installed so I installed it along with other packages that I found Here (I wanted to make sure that I will not see any other errors) and then there were no errors showed. But when I open ArcGIS I see the “City Energy Analyst” tool under “My Toolboxes” with a red X next to it and with no modules installed as you can see here:

![68aa1148-1568-11e7-9269-2a418147b50c](https://cloud.githubusercontent.com/assets/26455475/24602537/82b3d4a0-185d-11e7-86ad-490df165a4ac.PNG)

**Comment by daren-thomas** [comment]
when you right-click on the 'City Energy Analyst.pyt' file, you should find an option that shows you why it failed to load. What does it say?

**Comment by ahmed3yad** [comment]
@daren-thomas When I right-click, This menu appears: 

<img width="232" alt="1" src="https://cloud.githubusercontent.com/assets/26455475/24603199/2220a5d4-1860-11e7-8ff6-9091650a6b8d.png">

Which option of the above you are talking about?

**Comment by daren-thomas** [comment]
`Check Syntax...`

**Comment by ahmed3yad** [comment]
@daren-thomas 
<img width="368" alt="2" src="https://cloud.githubusercontent.com/assets/26455475/24603303/82acbc08-1860-11e7-99ae-e75ed13f6c15.png">

**Comment by daren-thomas** [comment]
huh... can you try to click `Refresh`?

**Comment by ahmed3yad** [comment]
@daren-thomas I did and still nothing happened ... This is the file named "[CityEnergyAnalyst.py](https://app.box.com/s/4slspyuv2jjnwyq1j7rikq2lh5bkxen9)" that is installed in the following directory: `C:\Users\Ayad\AppData\Roaming\ESRI\Desktop10.4\ArcToolbox\My Toolboxes\City Energy Analyst.pyt`

I do not know will it help or not that you check this file.

**Comment by daren-thomas** [comment]
@ahmed3yad could you do me a favor? could you delete the file `CityEnergyAnalyst.py` from the `My Toolboxes\City Energy Analyst.pyt` folder and retry the `cea install-toolbox` command?

If the folder appears again, please copy the `CityEnergyAnalyst.py` file to the `My Toolboxes` folder and delete the `City Energy Analyst.pyt` folder. Does it work now?

If the folder doesn't appear again, you should have a file `My Toolboxes\City Energy Analyst.pyt` (a file, not a folder!) - does that work?

**Comment by ahmed3yad** [comment]
@daren-thomas Great! .. now it works, but I had to copy the `CityEnergyAnalyst.py` file to the `My Toolboxes` folder and rename it to `CityEnergyAnalyst.pyt` and then deleting the `City Energy Analyst.pyt` folder. Deleting the file `CityEnergyAnalyst.py` and retry the `cea install-toolbox` command did not work and resulted in a `CityEnergyAnalyst.pyt` folder again and not a file.

I think now I have to test the tool with the `reference-case-open` scenario and I hope that it will work.

Thanks a lot for your help :)

**Comment by daren-thomas** [comment]
You're very welcome. Thank you for helping with the debugging: This annoying episode should be fixed by end of April!

**Comment by ahmed3yad** [comment]
@daren-thomas It seems that there are some more errors when running the `radiation` module
<img width="442" alt="1" src="https://cloud.githubusercontent.com/assets/26455475/24607985/af10502a-1874-11e7-8141-da6bb0b09e35.png">
<img width="831" alt="2" src="https://cloud.githubusercontent.com/assets/26455475/24607989/b1ae6fd8-1874-11e7-9a57-40653eb34335.png">

**Comment by daren-thomas** [comment]
@ahmed3yad where did you get the reference case from? can you use the command line to `cd` into a directory (e.g. `%temp%`) and then use `cea extract-reference-case` to extract the *reference-case-open* into that folder? then, while you're at it, `cd reference-case-open\baseline` from there and run `cea latitude` to extract the latitude value?

**Comment by ahmed3yad** [comment]
I got the reference case from your GitHub repository: `CEAforArcGIS\cea\examples\reference-case-open.zip`.

I extracted the latitude value, it is equal to 46.9524055556

**Comment by daren-thomas** [comment]
Can you `cd` into the folder you extracted the reference case to (`cea extract-reference-case` does just that, extract from that zip) and `activate cea` and then do `cea latitude`?

**Comment by ahmed3yad** [comment]
@daren-thomas I already did that and got the value of: 46.9524055556
<img width="399" alt="1" src="https://cloud.githubusercontent.com/assets/26455475/24613857/deb693f6-1889-11e7-8418-020203a203ad.png">
 
I also tried the tool with this extracted scenario and I had the same error.

**Comment by daren-thomas** [comment]
Oh dear. @ahmed3yad, please stand by while I try to figure out how to debug this...

**Comment by ahmed3yad** [comment]
@daren-thomas Of course! ... take your time

**Comment by daren-thomas** [comment]
@ahmed3yad: what time-zone are you in? I would like to contact tomorrow, maybe do a teamviewer session or similar.

**Comment by ahmed3yad** [comment]
@daren-thomas I'm currently staying in Italy, so my time zone is GMT+2 (the time now is 5:22 PM). It would be a pleasure to me to have such a contact with you, just tell me the time that most suits you and I will be available.

**Comment by daren-thomas** [comment]
@ahmed3yad I am in GMT+2 myself. Let us meet at 15:00.

**Comment by ahmed3yad** [comment]
@daren-thomas OK great! ... How will we do it? Skype, Facebook, TeamViewer??

**Comment by daren-thomas** [comment]
skype should work. send me an email to thomas (at) arch (dot) ethz (dot) ch with your skype name so I can call you.

**Comment by ahmed3yad** [comment]
@daren-thomas I have just sent you an email.

**Comment by daren-thomas** [comment]
@ahmed3yad try adding these two paths to your system environment PATH variable (https://superuser.com/questions/949560/how-do-i-set-system-environment-variables-in-windows-10):

- `C:\Users\YOUR_USER_NAME\Anaconda2\envs\cea`
- `C:\Users\YOUR_USER_NAME\Anaconda2\envs\cea\Library\bin`

I think that might do the trick for you.

**Comment by JIMENOFONSECA** [comment]
@ahmed3yad can we close this issue or do you need more assistance?

**Comment by ahmed3yad** [comment]
@daren-thomas Sorry for my late response. This is wonderful !, I tried what you said and the tool now works without any errors, I really appreciate your help. (NOTE: the `cea install-toolbox` command still creates a folder named `CityEnergyAnalyst.pyt` instead of a file and I have to manually get the `CityEnergyAnalyst.py` file out of that folder and rename it to `CityEnergyAnalyst.pyt`

**Comment by ahmed3yad** [comment]
@JIMENOFONSECA No, thanks a lot. The tool now works perfectly and you can close the issue. I just need some help on how to use the tool (Are there any detailed documentation for how to create a new scenario?)

**Comment by daren-thomas** [comment]
@ahmed3yad I will close this issue then. Thank you for pointing out the `cea install-toolbox` issue. I have fixed it in the current master version.

**Comment by ahmed3yad** [comment]
@daren-thomas  You're most welcome :)

**Comment by JIMENOFONSECA** [comment]
@ahmed3yad check our FAQ which includes information on this. please reasi an issue of something is not clear. thanks !
https://cityenergyanalyst.com/faq/

**Comment by ahmed3yad** [comment]
@JIMENOFONSECA Thanks a lot!, it seems like your "User Manual" has been updated 😃 ... I will check it and tell you If I find something that is not clear. (BTW, I think if you do some video tutorials it will be very helpful)
